ecl/hqlcpp/hqlcpp.cpp

```diff
@@ -7516,6 +7519,34 @@ void HqlCppTranslator::doBuildAssignExecuteWhen(BuildCtx & ctx, const CHqlBoundT
     }
 }
 
+void HqlCppTranslator::doBuildStmtExecuteWhen(BuildCtx & ctx, IHqlExpression * expr)
```

Comment:
Note: Compare with the previous function (which is almost identical, but evaluates a value instead).